src/hypothesis/stateful.py

```diff
@@ -518,3 +575,8 @@ def execute_step(self, step):
             )
             for target in rule.targets:
                 self.bundle(target).append(VarReference(name))
+
```

Comment:
It *definitely* wants to get run immediately after every rule. We want to check invariants after the last run.

It seems entirely reasonable to want to check invariants before the first rule, but it isn't obvious how to do that. It can't be done in `__init__`, since user code will probably run after that. I guess in `steps()` we could check if there are invariants and if this is the first run, and if so, return a marker step that just triggers running the invariants.